README.md

```diff
@@ -62,6 +62,114 @@ Loading this module through a script tag will make the `IpfsBitswap` object avai
 
 For the documentation see [API.md](API.md).
 
+### API
+
+#### `new Bitswap(libp2p, blockstore)`
+
+- `libp2p: Libp2p`, instance of the local network stack.
+- `blockstore: Blockstore`, instance of the local database (`IpfsRepo.blockstore`)
+
+Create a new instance.
+
+#### `getStream(cid)`
+
+- `cid: CID|Array`
+
+Returns a source `pull-stream`. Values emitted are the received blocks.
+
+Example:
+
+```js
+// Single block
+pull(
+  bitswap.getStream(cid),
+  pull.collect((err, blocks) => {
+    // blocks === [block]
+  })
+)
+
+// Many blocks
+pull(
+  bitswap.getStream([cid1, cid2, cid3]),
+  pull.collect((err, blocks) => {
+    // blocks === [block1, block2, block3]
+  })
+)
+```
+
+> Note: This is safe guarded so that the network is not asked
+> for blocks that are in the local `datastore`.
+
+#### `unwant(cids)`
+
+- `cids: CID|[]CID`
+
+Cancel previously requested keys, forcefully. That means they are removed from the
+wantlist independent of how many other resources requested these keys. Callbacks
+attached to `getBlock` are errored with `Error('manual unwant: key')`.
+
+#### `cancelWants(cids)`
+
+- `cid: CID|[]CID`
+
+Cancel previously requested keys.
+
+#### `putStream()`
+
+Returns a duplex `pull-stream` that emits an object `{key: Multihash}` for every written block when it was stored.
+Objects passed into here should be of the form `{data: Buffer, key: Multihash}`
```

Comment:
It is no long `data: Buffer, key Multihash`, instead it is `block: block, cid: cid`